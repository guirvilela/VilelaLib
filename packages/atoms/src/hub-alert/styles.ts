import styled from 'styled-components';

import { HubAlertProps } from './types';

export const AlertInnerStyle = styled.div`
  display: flex;
  gap: ${({ theme }) => theme.spaces[2]};
  margin-top: -2px;
  width: 100%;
`;

export const AlertTitleStyle = styled.span`
  font-family: ${({ theme }) => theme.typography.body2.family};
  font-size: ${({ theme }) => theme.typography.body2.size};
  line-height: ${({ theme }) => theme.typography.body2.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body2.letterSpacing};
  color: ${({ theme }) => theme.colors.neutral[800]};
`;

export const AlertContentStyle = styled.span`
  font-family: ${({ theme }) => theme.typography.body2.family};
  font-size: ${({ theme }) => theme.typography.body2.size};
  line-height: ${({ theme }) => theme.typography.body2.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body2.letterSpacing};
  color: ${({ theme }) => theme.colors.neutralGrey[700]};
  font-weight: 400;
  width: 100%;
  text-overflow: ellipsis;
  overflow: hidden;
`;

export const AlertCloseStyle = styled.button`
  background-color: transparent;
  width: ${({ theme }) => theme.spaces[3]};
  height: ${({ theme }) => theme.spaces[3]};
  border: none;
  color: inherit;
  cursor: pointer;
`;

export const AlertStyle = styled.div<Omit<HubAlertProps, 'title'>>`
  display: flex;
  width: 100%;
  background-color: ${({ theme, color }) =>
    color ? theme.colors[color][100] : theme.colors.neutral[100]};
  color: ${({ theme, color }) =>
    color ? theme.colors[color][500] : theme.colors.neutral[500]};
  border: 1px solid
    ${({ theme, color }) =>
      color ? theme.colors[color][500] : theme.colors.neutral[500]};
  border-radius: ${({ theme, radius }) =>
    radius ? theme.spaces[radius] : '4px'};
  align-items: ${({ inline }) => (inline ? 'center' : 'start')};

  ${({ theme, pad }) => {
    if (typeof pad === 'string') {
      return `padding: ${theme.spaces[pad]};`;
    }
    if (Array.isArray(pad)) {
      return `padding: ${pad.reduce(
        (acc, x) => acc + `${theme.spaces[x]} `,
        ''
      )};`;
    }
    return `padding: ${theme.spaces[2]};`;
  }}

  gap: ${({ theme, gap }) => (gap ? theme.spaces[gap] : theme.spaces[2])};

  ${AlertInnerStyle} {
    flex-direction: ${({ inline }) => (inline ? 'row' : 'column')};
    align-items: ${({ inline }) => (inline ? 'center' : 'start')};
  }

  ${AlertTitleStyle} {
    font-weight: ${({ inline }) => (inline ? 400 : 600)};
    white-space: ${({ nowrap }) => (nowrap ? 'nowrap' : undefined)};
  }
`;
