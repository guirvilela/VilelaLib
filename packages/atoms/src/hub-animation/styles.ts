import styled from 'styled-components';

import { HubAnimationProps } from './types';

export const AnimationStyle = styled.div<HubAnimationProps & { ref }>`
  ${({ theme, size, width, height }) => {
    switch (size) {
      case 'xs':
        return `
          width: ${theme.spaces[2]};
          height: ${theme.spaces[2]};
          min-width: ${theme.spaces[2]};
        `;
      case 'sm':
        return `
          width: ${theme.spaces[3]};
          height: ${theme.spaces[3]};
          min-width: ${theme.spaces[3]};
        `;
      case 'md':
        return `
          width: ${theme.spaces[5]};
          height: ${theme.spaces[5]};
          min-width: ${theme.spaces[5]};
        `;
      case 'lg':
        return `
          width: ${theme.spaces[6]};
          height: ${theme.spaces[6]};
          min-width: ${theme.spaces[6]};
        `;
      default:
        return `
          width: ${width || theme.spaces[5]};
          height: ${height || theme.spaces[5]};
          min-width: ${width || theme.spaces[5]};
        `;
    }
  }}
`;
