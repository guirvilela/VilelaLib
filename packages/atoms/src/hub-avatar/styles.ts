import styled from 'styled-components';

import { HubAvatarProps } from './types';

export const AvatarImgStyle = styled.img`
  width: 100%;
  height: 100%;
  background-size: cover;
  border-radius: 50%;
`;

export const AvatarInitialsStyle = styled.span`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  border-radius: 50%;
`;

export const AvatarStyle = styled.div<HubAvatarProps>`
  ${({ theme, size }) => {
    if (!size) {
      return `
        width: ${theme.spaces[7]};
        height: ${theme.spaces[7]};
        min-width: ${theme.spaces[7]};
      `;
    }

    switch (size) {
      case 'xs':
        return `
          width: ${theme.spaces[3]};
          height: ${theme.spaces[3]};
          min-width: ${theme.spaces[3]};
        `;
      case 'sm':
        return `
          width: ${theme.spaces[5]};
          height: ${theme.spaces[5]};
          min-width: ${theme.spaces[5]};
        `;
      case 'md':
        return `
          width: ${theme.spaces[7]};
          height: ${theme.spaces[7]};
          min-width: ${theme.spaces[7]};
        `;
      case 'lg':
        return `
          width: ${theme.spaces[10]};
          height: ${theme.spaces[10]};
          min-width: ${theme.spaces[10]};
        `;
      default:
        return `
          width: ${size};
          height: ${size};
          min-width: ${size};
        `;
    }
  }}
  ${({ borderless, theme }) =>
    borderless
      ? ''
      : `
    padding: ${theme.spaces[1]};
    border: 1px solid ${theme.colors.neutral[400]};
  `};

  border-radius: 50%;

  ${AvatarInitialsStyle} {
    ${({ theme, typo }) => `
      font-family: ${theme.typography[typo || 'h4'].family};
      font-size: ${theme.typography[typo || 'h4'].size};
      line-height: ${theme.typography[typo || 'h4'].lineHeight};
      letter-spacing: ${theme.typography[typo || 'h4'].letterSpacing};
    `}

    background-color: ${({ theme, color, colorLevel }) =>
      color && colorLevel
        ? theme.colors[color][colorLevel]
        : color && theme.colors[color]
        ? theme.colors[color][500]
        : color
        ? color
        : undefined};

    color: ${({ theme, colorText, colorTextLevel }) =>
      colorText && colorTextLevel
        ? theme.colors[colorText][colorTextLevel]
        : colorText && theme.colors[colorText]
        ? theme.colors[colorText][500]
        : colorText
        ? colorText
        : theme.colors.neutral[800]};
  }
`;
