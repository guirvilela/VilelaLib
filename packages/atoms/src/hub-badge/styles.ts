import styled, { css } from 'styled-components';

import { HubBadgeProps } from './types';

const allStyle = css<HubBadgeProps>`
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  gap: ${({ theme }) => theme.spaces[2]};

  ${({ theme, size, width, height }) => {
    if (width || height) {
      return `
        width: ${width};
        height: ${height};
      `;
    }

    switch (size) {
      case 'xs':
        return `
          width: ${theme.spaces[2]};
          height: ${theme.spaces[2]};
        `;
      case 'sm':
        return `
          width: ${theme.spaces[3]};
          height: ${theme.spaces[3]};
        `;
      case 'md':
        return `
          width: ${theme.spaces[4]};
          height: ${theme.spaces[4]};
        `;
      case 'lg':
        return `
          width: ${theme.spaces[5]};
          height: ${theme.spaces[5]};
        `;
      default:
        return '';
    }
  }}

  background-color: ${({ theme, color, colorLevel }) =>
    color && colorLevel
      ? theme.colors[color][colorLevel]
      : color
      ? theme.colors[color][100]
      : theme.colors.neutral[200]};

  color: ${({ theme, color, textColor, textColorLevel }) =>
    textColor && textColorLevel
      ? theme.colors[textColor][textColorLevel]
      : textColor
      ? theme.colors[textColor][500]
      : color
      ? theme.colors[color][500]
      : theme.colors.neutral[500]};

  border-radius: ${({ theme, rounded, radius }) =>
    rounded ? '50%' : theme.spaces[radius || '1']};

  ${({ theme, pad }) => {
    if (!pad) return `padding: 2px 6px`;
    if (typeof pad === 'string') return `padding: ${theme.spaces[pad]}`;
    if (pad.length === 4) {
      return `padding: ${theme.spaces[pad[0]]} ${theme.spaces[pad[1]]} ${
        theme.spaces[pad[2]]
      } ${theme.spaces[pad[3]]}`;
    }
    if (pad.length === 3) {
      return `padding: ${theme.spaces[pad[0]]} ${theme.spaces[pad[1]]} ${
        theme.spaces[pad[2]]
      }`;
    }
    if (pad.length === 2) {
      return `padding: ${theme.spaces[pad[0]]} ${theme.spaces[pad[1]]}`;
    }
    return `padding: ${theme.spaces[pad[0]]}`;
  }};

  font-family: ${({ theme }) => theme.typography.body2.family};
  font-size: ${({ theme }) => theme.typography.body2.size};
  line-height: ${({ theme }) => theme.typography.body2.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body2.letterSpacing};
  font-weight: 600;
`;

const floatStyle = css<HubBadgeProps>`
  position: relative;

  &::after {
    ${allStyle}

    position: absolute;
    content: '${({ content }) => content}';
    border: 1px solid ${({ theme }) => theme.colors.white[500]};
    ${({ theme, position }) => {
      if (position) {
        if (Array.isArray(position)) {
          return `
            top: -${theme.spaces[position[0]]};
            right: -${theme.spaces[position[1]]};
          `;
        }
        return `
          top: -${theme.spaces[position]};
          right: -${theme.spaces[position]};
        `;
      }
      return `
        top: -10px;
        right: -10px;
      `;
    }}
    ${({ theme }) => `
      min-width: ${theme.spaces[4]};
      width: ${theme.spaces[4]};
      height: ${theme.spaces[4]};
    `}
    padding: 0;

    font-family: ${({ theme }) => theme.typography.body3.family};
    font-size: ${({ theme }) => theme.typography.body3.size};
    line-height: ${({ theme }) => theme.typography.body3.lineHeight};
    letter-spacing: ${({ theme }) => theme.typography.body3.letterSpacing};
  }
`;

export const BadgeStyle = styled.div<HubBadgeProps>`
  ${({ float, content }) => {
    if (float) {
      if (content) {
        return floatStyle;
      }
    } else {
      return allStyle;
    }
    return '';
  }}
`;
