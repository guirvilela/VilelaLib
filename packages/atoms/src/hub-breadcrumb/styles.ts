import styled from 'styled-components';

import { HubBreadcrumbProps, HubBreadcrumbsProps } from './types';

export const BreadcrumbsStyle = styled.div<HubBreadcrumbsProps>`
  display: flex;
  align-items: center;
  overflow: auto;
`;

export const BreadcrumbsPathStyle = styled.div`
  display: flex;
  align-items: center;
`;

export const BreadcrumbsIconStyle = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  color: ${({ theme }) => theme.colors.neutral[400]};
  margin: ${({ theme }) => `0 ${theme.spaces[2]}`};
`;

export const BreadcrumbStyle = styled.span<HubBreadcrumbProps>`
  display: flex;
  align-items: center;
  gap: ${({ theme }) => theme.spaces[2]};
  font-family: ${({ theme }) => theme.typography.body2.family};
  font-size: ${({ theme }) => theme.typography.body2.size};
  line-height: ${({ theme }) => theme.typography.body2.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body2.letterSpacing};
  color: ${({ theme, active }) =>
    active ? theme.colors.primary[500] : theme.colors.neutral[400]};
  transition: color 0.5s;
  font-weight: 600;
  white-space: ${({ nowrap }) => (nowrap ? 'nowrap' : 'pre-wrap')};

  ${({ width }) =>
    width
      ? `
    display: block;
    max-width: ${width};
    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;
  `
      : ''}

  &:hover {
    color: ${({ theme }) => theme.colors.primary[500]};
  }
`;
