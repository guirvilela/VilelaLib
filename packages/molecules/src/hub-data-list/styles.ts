import { HubList } from '@hubert/atoms';
import styled from 'styled-components';

export const ListStyle = styled(HubList)`
  @media print {
    max-height: unset;
    height: auto;
    overflow: unset;
  }
`;
