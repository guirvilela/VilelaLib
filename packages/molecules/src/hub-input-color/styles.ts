import styled from 'styled-components';

export const InputColor = styled.input.attrs({
  type: 'color'
})`
  background-color: transparent;
  border: none;
  width: ${({ theme }) => theme.spaces[6]};
  border-radius: ${({ theme }) => theme.spaces[2]};
`;
