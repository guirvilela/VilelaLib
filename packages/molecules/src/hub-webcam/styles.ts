import styled from 'styled-components';

export const VideoContainer = styled.video`
  @media (${({ theme }) => theme.mediaQuery.md}) {
    width: 100%;
  }
`;

export const PreviewContainer = styled.canvas`
  @media (${({ theme }) => theme.mediaQuery.md}) {
    width: 100%;
  }
`;
