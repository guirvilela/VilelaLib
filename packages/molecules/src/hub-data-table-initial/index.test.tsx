import { I18nProvider } from '@hubert/i18n';
import { wrapse } from '@hubert/shared';
import { ThemeProvider } from '@hubert/styles';
import { render } from '@testing-library/react';
import React from 'react';

import { HubDataTableInitial } from '.';

describe('HubDataTableInitial component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <HubDataTableInitial title={'aaaa'} content={'bbbbb'} />,
      {
        wrapper: wrapse(ThemeProvider, I18nProvider)
      }
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
