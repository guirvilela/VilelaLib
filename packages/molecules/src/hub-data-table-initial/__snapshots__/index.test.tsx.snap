// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HubDataTableInitial component matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  gap: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  padding: 48px 20px;
}

.c1 {
  gap: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 540px;
}

.c3 {
  gap: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.c4 {
  font-family: Kanit;
  font-size: 32px;
  line-height: 1.2666666666666666;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
  font-weight: 500;
  text-align: center;
  color: #083350;
  text-overflow: ellipsis;
  overflow: hidden;
}

.c5 {
  font-family: Nunito Sans;
  font-size: 18px;
  line-height: 1.5;
  -webkit-letter-spacing: 0.01em;
  -moz-letter-spacing: 0.01em;
  -ms-letter-spacing: 0.01em;
  letter-spacing: 0.01em;
  font-weight: 600;
  text-align: center;
  color: #304A5D;
  text-overflow: ellipsis;
  overflow: hidden;
}

.c2 {
  color: #083350 !important;
}

@media (max-width:768px) {
  .c1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    width: 100%;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
      width="540px"
    >
      <svg
        class="c2"
        fill="none"
        height="144"
        viewBox="0 0 144 144"
        width="144"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="72"
          cy="72"
          fill="#E8EEF4"
          r="72"
        />
        <path
          d="M68.1952 46.2437C57.5267 46.2437 48.8781 54.8923 48.8781 65.5608C48.8781 76.2293 57.5267 84.8779 68.1952 84.8779C78.8637 84.8779 87.5123 76.2293 87.5123 65.5608C87.5123 54.8923 78.8637 46.2437 68.1952 46.2437ZM42.4391 65.5608C42.4391 51.3361 53.9705 39.8047 68.1952 39.8047C82.4199 39.8047 93.9513 51.3361 93.9513 65.5608C93.9513 71.5127 91.9324 76.9931 88.542 81.3546L105.886 98.6989C107.144 99.9562 107.144 101.995 105.886 103.252C104.629 104.509 102.591 104.509 101.333 103.252L83.989 85.9076C79.6275 89.298 74.1471 91.3169 68.1952 91.3169C53.9705 91.3169 42.4391 79.7855 42.4391 65.5608Z"
          fill="#083350"
        />
      </svg>
      <div
        class="c3"
      >
        <h2
          class="c4"
          color="primary"
        >
          aaaa
        </h2>
        <span
          class="c5"
          color="neutral"
        >
          bbbbb
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
